### ISSUE TITLE
Paper Mario - Some nice and sharp; Some blurry & badly processed (Textures)

### ISSUE THREAD

**Issue by maximemoring** [gonetz/GLideN64#2082]
I'm having this issue where in the game, some text of the same type will be perfectly sharp and some other will be either blurry or badly processed like I was using a bad texture enhancer when I'm not.

Tried enabling and unabling every single option, changed resolutions, refresh rates, and it seems like using a texture enhancer is the only thing that overwrites the poor default texture handling

Using latest GLideN64 (Public Release 4.0) on Project64 2.2.0.3
Only using multiples of native resolution (see screenshot, currently using native x4)

![image](https://user-images.githubusercontent.com/3126740/63652890-b4144e80-c733-11e9-8ad1-7aac7df25df8.png)
You can see here the perfect example. 
Blurry: The "x" and numbers show up perfectly sharp, but everything else doesn't
Badly processed: The numbers show up perfectly sharp yet the "/" and letters don't, and end up with this extra wide, wierdly high resolution border. 

Options:
![image](https://user-images.githubusercontent.com/3126740/63652963-c8a51680-c734-11e9-8925-e8d5fa207bc6.png)
![image](https://user-images.githubusercontent.com/3126740/63652968-d5296f00-c734-11e9-9f12-89e207f3b9e5.png)
![image](https://user-images.githubusercontent.com/3126740/63652977-e4102180-c734-11e9-8dde-90197ac8a94b.png)
![image](https://user-images.githubusercontent.com/3126740/63652983-ed00f300-c734-11e9-8c57-2078f677b628.png) 
Texture packs is also off and never touched OSD

**Comment by oddMLan** [comment]
That's why I wish there was an option to force point-filtering.

**Comment by maximemoring** [comment]
Is point filtering some kind of embedded texture enhancement?

**Comment by oddMLan** [comment]
No, it was an option that the predecesor to this plugin (Glide64) had. Basically it would force all textures to use nearest neighbor instead of bilinear. Including those that are normally filtered on console.
There's a ticket report already open for it #574

**Comment by maximemoring** [comment]
Oh wow, 4 years ago...

**Comment by gonetz** [comment]
oddMLan is right in one point - blurry textures are filtered ones; sharp textures are not filtered.
Proper emulation of texture filtering is tricky (impossible?) in non-native resolutions.

**Comment by Jj0YzL5nvJ** [comment]
Try `Enable halos removal`. That improves the text in some cases.

**Comment by gonetz** [comment]
Source of problems with texture filtering in hi-res is obvious: small texture was designed to be applied to low-res polygon, but actually is used for large polygon. For point-sampling it is not a problem - in case of multiples of native res we will just get larger pixels. But filtering works differently for small and for large polygons, especially on image borders. The smaller a texture, the more chances that the result will look somehow wrong.

> There is a line in the .ini file that is called texture\bilinearMode= with default "1"

This setting switches between normal 4-points bilinear filtering and 3-points one used by N64 (bilinearMode=0). The difference is not obvious but visible.

> Is it really too much to ask to have the option to not use bilinear filtering?

No, it is not hard. But it is hack, and I dislike to add more hacks.

**Comment by darkychao** [comment]
> No, it was an option that the predecesor to this plugin (Glide64) had.
Semi-correction, you actually could also do this in earlier builds of GLideN64. There was no way to set it in the GUI but using "BilinearMode=3" would force nearest neighbor. Sometime between [597f7a2](https://github.com/gonetz/GLideN64/commit/597f7a25cfe2f9ed4c1474daef1d22027e32fcb8) and [c5c0fbe](https://github.com/gonetz/GLideN64/commit/c5c0fbeb6fe3ea6b0de8c6353910dc6bdb038ec5) something changed and this stopped working.

**Comment by maximemoring** [comment]
I would like to ask more clearly for that option to be available, not necessarily on the interface, as I can understand the complaints about things not looking right with it on might be annoying, but at the very least have that `BilinearMode=3` work, or something similar.

**Comment by Papermanzero** [comment]
The mish mash in 2D games is really annoying.
There is already a long discussion about it here:
https://github.com/gonetz/GLideN64/issues/1788

**Comment by maximemoring** [comment]
Until this is done, I will be using Angrylion's RDP Plus plugin, as with an AMD Ryzen 7 2700X with 16 threads, it's a walk in the park. (PJ64 uses at most 30% of CPU time) It does exactly what I want which is point filtering while seemingly forcing a multiple of native resolution when maximizing the window.

Yes, that plugin doesn't scale render resolution, and I wish I could have it both ways, but until point/nearest neighbor filtering is available, this won't be possible and I'd rather have a fully pixelated look than a half-and-half blurry mess.

**Comment by Nomadsx** [comment]
I'd also really like to see nearest neighbor added. Bilinear is the first thing I disable in literally every emulator since it just smudges everything. Being forced to use it with this nice plugin is a shame. Angrylion's RDP Plus puts out a great sharp image but even with the latest build, it's too taxing with certain games and lacks a lot of features of this plugin.

**Comment by AlterSpectrum** [comment]
I'd also like to add that I would love for this option to be implemented, can't stand blurry textures or bilinear filtering. I understand having filtering is the authentic experience but the option for those that would like to emulate games totally unfiltered would be extremely appreciated. All GLideN64 needs is an option for force point-sampled filtering like Glide64 had. Games like Mischief Makers and Paper Mario suffer greatly from the inconsistency of blurred and sharp textures. Just makes the whole experience look very ugly.

Here's Mischief Makers running with force point-sampled on Glide64 Final. Looks how it should, with sharp unfiltered pixels on all objects and surfaces:
https://i.imgur.com/UGjCpts.png

But the problem using Glide64 is you cant crop or overscan the image. So for games like Mischief Maker you are stuck with a pretty small display area with big black bars all around.

Here it is on GLideN64 4.0 using N64 style 3 point. A bunch of objects and surfaces have ugly blurred textures with visible tile seams:
https://i.imgur.com/FvBThbz.png

Now I did manage to randomly find an old build of GLideN64 (rev.365838e) that let me disable the filtering. This allowed me to crop the image a bit to get a slightly larger display area. However the cropping method in the older builds is nowhere near as versatile as the overscan method in 4.0 which lets you cut out any unwanted screen space with total control.

So basically in an ideal world we would have an updated version of 4.0 with the option to turn off all filtering and/or enable force point-sampled filtering on all objects and surfaces. I've seen quite a few people requesting this who also hate the blurry smeared look of bilinear filtering so it would really be appreciated.

Besides this one snag it’s a fantastic plugin that easily provides the best graphical experience by far!
